lib/lock.js

```diff
@@ -42,9 +43,17 @@ exports.reapLock = function(key, callback) {
  */
 exports.acquireLock = function(key, maximumLocks, ttlSeconds, callback) {
   var lockId = uuid.v1();
-  client.eval(lua.acquire_lock, 1, key, lockId, maximumLocks, ttlSeconds, function(err, response) {
+  async.waterfall([
+    function(cb) { return redis.ensureScript(client, lua.acquire_lock, cb); },
+    function(cb) {
+      client.eval(lua.acquire_lock, 1, key, lockId, maximumLocks, ttlSeconds, function(err, response) {
+        if (err) { return cb(err, null); }
+        return cb(null, response ? lockId : null);
+      });
+    }
+  ], function(err, lockId) {
     if (err) { return callback(err, null); }
-    return callback(null, response ? lockId : null);
+    return callback(null, lockId);
```

Comment:
I *think* you can just pass `callback` straight to `waterfall` here. The Inline function seems unnecessary.